src/Discord.Net.WebSocket/DiscordSocketClient.cs

```diff
@@ -627,30 +631,36 @@ private async Task ProcessUserDownloadsAsync(IEnumerable<SocketGuild> guilds)
         {
             var cachedGuilds = guilds.ToImmutableArray();
 
-            const short batchSize = 1;
-            ulong[] batchIds = new ulong[Math.Min(batchSize, cachedGuilds.Length)];
-            Task[] batchTasks = new Task[batchIds.Length];
-            int batchCount = (cachedGuilds.Length + (batchSize - 1)) / batchSize;
+            foreach (var guild in cachedGuilds)
+            {
+                await ApiClient.SendRequestMembersAsync(guild.Id).ConfigureAwait(false);
+                await guild.DownloaderPromise.ConfigureAwait(false);
+            }
+        }
 
-            for (int i = 0, k = 0; i < batchCount; i++)
+        /// <inheritdoc />
+        public override async Task DownloadUsersAsync(IGuild guild, IEnumerable<ulong> userIds)
+        {
+            if (ConnectionState == ConnectionState.Connected)
             {
-                bool isLast = i == batchCount - 1;
-                int count = isLast ? (cachedGuilds.Length - (batchCount - 1) * batchSize) : batchSize;
+                EnsureGatewayIntent(GatewayIntents.GuildMembers);
 
-                for (int j = 0; j < count; j++, k++)
+                var socketGuild = GetGuild(guild.Id);
+                if (socketGuild != null)
                 {
-                    var guild = cachedGuilds[k];
-                    batchIds[j] = guild.Id;
-                    batchTasks[j] = guild.DownloaderPromise;
+                    await ProcessUserDownloadsAsync(socketGuild, userIds).ConfigureAwait(false);
                 }
+            }
```

Comment:
This should log an error if the connection state is not `Connected` instead of silently failing

Reply:
I can add that, I just oriented myself on what the existing `DownloadUsersAsync(IEnumerable<IGuild> guilds)` is doing, which also silently fails.
What would you recommend? Throwing an exception or logging an error?

Reply:
I'd say throwing an exception would be better in this case